### PR TITLE
Initial guidelines

### PR DESCRIPTION
**Pull request by ryyppy**
Ready for review! (@astuetz @Aaang)

### CODE REVIEW

**Comment by ghost** [comment]
LGTM

**Comment by ryyppy** [comment]
Thanks to @busypeoples for initial proof reading :+1: 🎉

**Comment by busypeoples** [comment]
@ryyppy Great writeup, excellent work here. This will be very beneficial for on-boarding other teams as well.

File: guidelines/README.md [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+  ```javascript
+  // bad
+  type foo = string;
+  type myFoo = string;
+  type JSONApiEntity = {};
+
+  // good
+  type Foo = string;
+  type MyFoo = string;
+  type JsonApiEntity = {};
```
Comment (nikgraf) [inline guidelines/README.md:44]: love it ❤️

Comment (mxstbr) [reply guidelines/README.md:44]: Ugh

Comment (Aaang) [reply guidelines/README.md:44]: TBH I'm usually in favor of writing acronyms in uppercase, so `JSONAPIEntity` in this case, but in this specific case it really looks weird but still it is more human readable.

Acronyms not written in uppercase can end up with a completely different meaning.
for instance, you want to load the members of the Austrian League of Legends (ALL - fictional):
`loadAllMembers` vs. `loadALLMembers`

Comment (ryyppy) [reply guidelines/README.md:44]: It's too hard to argue about this topic, it's bike-shedding on it's finest, so I will just leave it as it is right now and we will correct this after a proper evaluation.... In our current codebases, I barely used this acronym convention :-(

Comment (Aaang) [reply guidelines/README.md:44]: You're right, fine for me too!
I just wanted to point it out that acronyms might get misleading if not written in upper case, even though variable names are better readable in most cases when considered normal words.

**Pull request merged**